crates/subspace-networking/src/node.rs

```diff
@@ -1,19 +1,90 @@
-use crate::shared::{Command, Shared};
+use crate::shared::{Command, CreatedSubscription, Shared};
+use bytes::Bytes;
 use event_listener_primitives::HandlerId;
-use futures::channel::oneshot;
+use futures::channel::{mpsc, oneshot};
 use futures::SinkExt;
 use libp2p::core::multihash::Multihash;
+use libp2p::gossipsub::error::SubscriptionError;
+use libp2p::gossipsub::Sha256Topic;
 use libp2p::{Multiaddr, PeerId};
+use std::ops::{Deref, DerefMut};
 use std::sync::Arc;
 use thiserror::Error;
 
+/// Topic subscription, will unsubscribe when last instance is dropped for a particular topic.
+#[derive(Debug)]
+pub struct TopicSubscription {
+    topic: Option<Sha256Topic>,
+    subscription_id: usize,
+    command_sender: Option<mpsc::Sender<Command>>,
+    receiver: mpsc::UnboundedReceiver<Bytes>,
+}
+
+impl Deref for TopicSubscription {
+    type Target = mpsc::UnboundedReceiver<Bytes>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.receiver
+    }
+}
+
+impl DerefMut for TopicSubscription {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.receiver
+    }
+}
```

Comment:
Also `derive_more` has derives for [`Deref`](https://jeltef.github.io/derive_more/derive_more/deref.html) and [`DerefMut`](https://jeltef.github.io/derive_more/derive_more/deref_mut.html).